insights/parsers/tests/test_ldif_config.py

```diff
@@ -0,0 +1,174 @@
+ er || nsds5replicaname || nsds5replicaport || nsds5replicaprotocoltimeout ||
+ nsds5replicapurgedelay || nsds5replicareferral || nsds5replicaroot || nsds5re
+ plicasessionpausetime || nsds5replicastripattrs || nsds5replicatedattributeli
+ st || nsds5replicatedattributelisttotal || nsds5replicatimeout || nsds5replic
+ atombstonepurgeinterval || nsds5replicatransportinfo || nsds5replicatype || n
+ sds5replicaupdateinprogress || nsds5replicaupdateschedule || nsds5task || nsd
+ s7directoryreplicasubtree || nsds7dirsynccookie || nsds7newwingroupsyncenable
+ d || nsds7newwinusersyncenabled || nsds7windowsdomain || nsds7windowsreplicas
+ ubtree || nsruvreplicalastmodified || nsstate || objectclass || onewaysync ||
+  winsyncdirectoryfilter || winsyncinterval || winsyncmoveaction || winsyncsub
+ treepair || winsyncwindowsfilter")(targetfilter = "(|(objectclass=nsds5Replic
+ a)(objectclass=nsds5replicationagreement)(objectclass=nsDSWindowsReplicationA
+ greement)(objectClass=nsMappingTree))")(version 3.0;acl "permission:Read Repl
+ ication Agreements";allow (compare,read,search) groupdn = "ldap:///cn=Read Re
+ plication Agreements,cn=permissions,cn=pbac,dc=idm,dc=nypd,dc=finest";)
+aci: (targetattr=*)(version 3.0;acl "permission:Add Replication Agreements";al
+ low (add) groupdn = "ldap:///cn=Add Replication Agreements,cn=permissions,cn=
+ pbac,dc=idm,dc=nypd,dc=finest";)
+aci: (targetattr=*)(targetfilter="(|(objectclass=nsds5Replica)(objectclass=nsd
+ s5replicationagreement)(objectclass=nsDSWindowsReplicationAgreement)(objectCl
+ ass=nsMappingTree))")(version 3.0; acl "permission:Modify Replication Agreeme
+ nts"; allow (read, write, search) groupdn = "ldap:///cn=Modify Replication Ag
+ reements,cn=permissions,cn=pbac,dc=idm,dc=nypd,dc=finest";)
+aci: (targetattr=*)(targetfilter="(|(objectclass=nsds5replicationagreement)(ob
+ jectclass=nsDSWindowsReplicationAgreement))")(version 3.0;acl "permission:Rem
+ ove Replication Agreements";allow (delete) groupdn = "ldap:///cn=Remove Repli
+ cation Agreements,cn=permissions,cn=pbac,dc=idm,dc=nypd,dc=finest";)
+aci: (target = "ldap:///cn=meTo($dn),cn=*,cn=mapping tree,cn=config")(targetat
+ tr = "objectclass || cn")(version 3.0; acl "Allow hosts to read their replica
+ tion agreements"; allow(read, search, compare) userdn = "ldap:///fqdn=($dn),c
+ n=computers,cn=accounts,dc=idm,dc=nypd,dc=finest";)
+cn: mapping tree
+modifiersName: cn=Directory Manager
+modifyTimestamp: 20201026161600Z
+objectClass: top
+objectClass: extensibleObject
+numSubordinates: 3
+
+dn: cn=sasl,cn=config
+cn: sasl
+objectClass: top
+objectClass: nsContainer
+numSubordinates: 1
+"""
+
+LDIF_CONFIG_EMPTY = ""
+
+
+def test_ldip_parser():
+    ldif_config = LDIFParser(context_wrap(LDIF_CONFIG))
+    assert ldif_config[1]['cn=config']['modifiersName'] == 'cn=directory manager'
+    assert ldif_config[1]['cn=config']['modifyTimestamp'] == '20210609192548Z'
+    assert ldif_config[1]['cn=config']['objectClass'] == 'nsslapdConfig'
```

Comment:
Please add more tests for other lines and keys. e.g. the `aci`, `cn`

Reply:
Added more test cases.

Reply:
I don't see the test for the `aci:` yet, which is the most complex one in the parser. Please add them as well.

BTW, please ensure the coverage test of the parser reaches 100%

Thanks.

Reply:
@xiangce - Updated the code for it, please have a check. Thanks.